chiselFrontend/src/main/scala/chisel3/core/Mem.scala

```diff
@@ -143,7 +143,7 @@ object SeqMem {
   * @note when multiple conflicting writes are performed on a Mem element, the
   * result is undefined (unlike Vec, where the last assignment wins)
   */
-sealed class SeqMem[T <: Data](t: T, n: Int) extends MemBase[T](t, n) {
+sealed class SyncReadMem[T <: Data](t: T, n: Int) extends MemBase[T](t, n) {
   def read(addr: UInt, enable: Bool): T = {
```

Comment:
It does!
```scala
scala> class Foo(val a: Int)
defined class Foo

scala> type Bar = Foo
defined type alias Bar

scala> val bar = new Bar(3)
bar: Foo = Foo@33433fc5

scala> bar.isInstanceOf[Bar]
res0: Boolean = true

scala> bar match {
     | case _: Bar => print("Success!")
     | }
Success!
```
It doesn't work if you try to do reflection though:
```scala
scala> bar.getClass
res5: Class[_ <: Foo] = class Foo
```
But this is probably okay